TombEngine/Objects/Generic/Switches/generic_switch.cpp

```diff
@@ -63,85 +76,115 @@ namespace TEN::Entities::Switches
 		{
 			auto bounds = GameBoundingBox(switchItem);
 
-			if ((switchItem->TriggerFlags == 3 || switchItem->TriggerFlags == 4) && switchItem->Animation.ActiveState == SWITCH_ON)
+			if ((switchItem->TriggerFlags == 3 || switchItem->TriggerFlags == 4) && switchItem->Animation.ActiveState == SWITCH_OFF)
 				return;
 
 			SwitchBounds.BoundingBox.X1 = bounds.X1 - 256;
 			SwitchBounds.BoundingBox.X2 = bounds.X2 + 256;
 
-			if (switchItem->TriggerFlags)
+			SwitchBounds.BoundingBox.Z1 = bounds.Z1 - 512;
+			SwitchBounds.BoundingBox.Z2 = bounds.Z2 + 512;
+
+			switch (switchItem->TriggerFlags)
```

Comment:
Maybe it will be possible to define an `ObjectsCollisionBounds` object for each switch?

Reply:
I can try filled them also in the next case, like the SwitchPos.Z

Although I'm not sure which values, most of them could be similar (except for the big button and giant button that yes require a bigger bound like 512 one).

I'll update them and left the rest to 256 to see how it feels.

Reply:
At the end I made some more changes:
-  Some near of the wall are at BLOCK(0.2)
- Others that have more volume (like valve or small lever) I let them at BLOCK(0.25)
- The big button and giant button uses BLOCK(0.5) because Lara must get far from the wall to do the animation without digging in the object.
- For the custom at the moment I let the BLOCK(0.5), I don't know which animation will the builder use, so I let the big area, still in the future, if that happens to be insufficient, I could use the ItemFlags[7] for that, but I would prefer not to.